cloudinit/controlplane_join.go

```diff
@@ -39,19 +39,13 @@ type ControlPlaneJoinInput struct {
 	BaseUserData
 	Certificates
 
-	BootstrapToken      string
-	ControlPlaneAddress string
-	JoinConfiguration   string
+	BootstrapToken    string
+	JoinConfiguration string
 }
 
 // NewJoinControlPlane returns the user data string to be used on a new control plane instance.
 func NewJoinControlPlane(input *ControlPlaneJoinInput) ([]byte, error) {
 	input.Header = cloudConfigHeader
-	if err := input.Certificates.validate(); err != nil {
-		return nil, errors.Wrapf(err, "ControlPlaneInput is invalid")
-	}
-
-	input.WriteFiles = certificatesToFiles(input.Certificates)
```

Comment:
Are we intending to drop support for k8s/kubeadm v1.13 here? Otherwise we'd still need to write the certificates to disk for a control plane join.

Reply:
good point. it is still not clear to me what is the expected flow for certificates, I have to sync with @ncdc @chuckha ...

Reply:
I'm not up to speed on <=1.13 vs >= 1.14 & kubeadm & certs. Can someone please fill me in?

Reply:
v1.14+ it is possible to do a kubeadm join without manually copying the certificates using by using a encrypted and timeboxed secret

Reply:
My impression of using the --certificate-upload flag was that it was for human users and that automating it would be more of a pain than automating the cert creation.

My expected workflow was => on create or join look for certificate secrets. If they exist then write them to disk (/etc/kubernetes/pki). If they don't exist then generate the CAs and upload those to secrets (should only happen with first control plane). It doesn't feel very secure since secrets aren't particularly secret.

Reply:
@chuckha since we already have the secrets we could basically replicate the --certificate-upload process as part of the bootstrapping as we do today for the bootstrap tokens.

Reply:
kk. I restored the deleted lines and isolated certificate generation in a separated func, so we can plugin the certificate flow described above in a follow-up PR